data_diff/database.py

```diff
@@ -212,23 +247,48 @@ def query(self, sql_ast: SqlOrStr, res_type: type):
     def enable_interactive(self):
         self._interactive = True
 
-    def _parse_type(self, type_repr: str, datetime_precision: int = None, numeric_precision: int = None) -> ColType:
+    def _convert_db_precision_to_digits(self, p: int) -> int:
+        """Convert from binary precision, used by floats, to decimal precision."""
+        # See: https://en.wikipedia.org/wiki/Single-precision_floating-point_format
+        return math.floor(math.log(2**p, 10))
+
+    def _parse_type(
+        self, type_repr: str, datetime_precision: int = None, numeric_precision: int = None, numeric_scale: int = None
+    ) -> ColType:
         """ """
 
         cls = self.DATETIME_TYPES.get(type_repr)
         if cls:
             return cls(
-                precision=datetime_precision if datetime_precision is not None else DEFAULT_PRECISION,
+                precision=datetime_precision if datetime_precision is not None else DEFAULT_DATETIME_PRECISION,
                 rounds=self.ROUNDS_ON_PREC_LOSS,
             )
 
+        cls = self.NUMERIC_TYPES.get(type_repr)
+        if cls:
+            if issubclass(cls, Integer):
+                # Some DBs have a constant numeric_scale, so they don't report it.
+                # We fill in the constant, so we need to ignore it for integers.
+                return cls(precision=0)
+
+            elif issubclass(cls, Decimal):
+                return cls(precision=numeric_scale)
+
+            assert issubclass(cls, Float)
+            # assert numeric_scale is None
+            return cls(
+                precision=self._convert_db_precision_to_digits(
+                    numeric_precision if numeric_precision is not None else DEFAULT_NUMERIC_PRECISION
```

Comment:
Why is 4-bytes float the safe default? Because even if it's 8-bytes, it is "at least" 4-bytes? 

Isn't this only a safe default if it's a float4 and `numeric_precision` isn't in the schema? This seems like code that should be in the driver it's relevant for, not the abstract class, no?

Should we maybe introduce `Double` in addition, so they can be typed out in `NUMERIC_TYPES` for each class?

Reply:
Yes.

> not the abstract class

See the refactor PR, it addresses this concern.

> Should we maybe introduce Double in addition

It isn't necessary for most DBs, because they provide numeric_scale.

For those that don't, yes, it can give us more accuracy. It just doesn't seem like high priority.

Reply:
👍🏻 I think it's worth a comment why this is a safe default, but yeah, I agree with you